### ISSUE TITLE
Rename QuantumCircuit.num_nonlocal_gates to num_multiqubit_gates

### ISSUE THREAD

**Issue by kevinsung** [Qiskit/qiskit-terra#10565]
### What should we add?

My understanding is that a "nonlocal" gate typically refers to a gate that involves qubits that are not connected on the hardware graph. The method `QuantumCircuit.num_nonlocal_gates` actually computes the number of multi-qubit gates, so it should be called `num_multiqubit_gates`. The source code for the function even uses `multi_qubit_gates` for the variable name.
https://github.com/Qiskit/qiskit-terra/blob/c29e938c2ba1c359a0a3268679c68f3237b2c1b7/qiskit/circuit/quantumcircuit.py#L2065-L2076

**Comment by Cryoris** [comment]
I agree `multiqubit` would be more precise. However, looking at this function, do we actually need it? We could also maybe enhance `count_ops` to accept a filter function (similar to `depth`) to do this. Already in it's current state, `count_ops` probably covers almost all use cases where you'd use `num_nonlocal_gates`.
